Lib9c/Action/ClaimStakeReward.cs

```diff
@@ -247,81 +349,138 @@ public override IAccountStateDelta Execute(IActionContext context)
                     avatarState.inventory.Serialize());
         }
 
+        private static List<StakeRegularFixedRewardSheet.RewardInfo> GetRegularFixedRewardInfos(
+            IAccountState states,
+            int level)
+        {
+            return states.TryGetSheet<StakeRegularFixedRewardSheet>(out var fixedRewardSheet)
+                ? fixedRewardSheet[level].Rewards
+                : new List<StakeRegularFixedRewardSheet.RewardInfo>();
+        }
+
         private IAccountStateDelta ProcessReward(
             IActionContext context,
             IAccountStateDelta states,
             ref AvatarState avatarState,
             ItemSheet itemSheet,
-            FungibleAssetValue stakedAmount,
+            FungibleAssetValue stakedFav,
             int itemRewardStep,
             int runeRewardStep,
             int currencyRewardStep,
+            int currencyCrystalRewardStep,
             List<StakeRegularFixedRewardSheet.RewardInfo> fixedReward,
             List<StakeRegularRewardSheet.RewardInfo> regularReward)
         {
-            var stakedCurrency = stakedAmount.Currency;
-
             // Regular Reward
             foreach (var reward in regularReward)
             {
+                if (reward.DecimalRate is null)
+                {
+                    throw new NullReferenceException(
+                        "decimal rate is null." +
+                        $" \"{ActionTypeText}\" action requires to set decimal rate.");
+                }
+
+                var stakedQuantity = decimal.Parse(
+                    stakedFav.GetQuantityString(),
+                    CultureInfo.InvariantCulture);
+                var rewardQuantityForSingleStep =
+                    Math.Truncate(stakedQuantity / reward.DecimalRate.Value);
```

Comment:
I think that converting `reward.DecimalRate` to FAV with proper currency (atm NCG) would be better than converting FAV to decimal...